packages/gotrue/lib/src/types/mfa.dart

```diff
@@ -277,9 +296,10 @@ class AMREntry {
 
   factory AMREntry.fromJson(Map<String, dynamic> json) {
     return AMREntry(
-      method: AMRMethod.values.firstWhere(
-        (e) => e.name == json['method'],
-      ),
+      method: AMRMethod.values.firstWhereOrNull(
+            (e) => e.code == json['method'],
+          ) ??
+          AMRMethod.unknown,
```

Comment:
It is possible to use `orElse` instead of `collection.firstWhereOrNull`.

```suggestion
      method: AMRMethod.values.firstWhere(
            (e) => e.code == json['method'],
            orElse: () => AMRMethod.unknown
          ),
```